shopfloor_manual_product_transfer/services/manual_product_transfer.py

```diff
@@ -511,9 +521,17 @@ def _unreserve_other_lines(self, location, move_lines):
             [
                 ("location_id", "=", location.id),
                 ("state", "in", ("assigned", "partially_available")),
+                # FIXME: maybe the answer to the TODO below: should we not
+                # add 'product_id' in the domain here? We could block the entire
+                # process here (see check on 'extra_move_lines' below) because
+                # another move line for another product is found in the given
+                # location... is it what we want?
             ]
         )
         extra_move_lines = location_move_lines - move_lines
+        # TODO: question, why do we care about lines that could be related to
+        # other products here? Should it prevent us to unreserve the moves of
+        # the product we're interested in?
```

Comment:
ping @jbaudoux I would like your insight on this, I think `location_move_lines` could contain unrelated lines (of other products sharing the same location), maybe it doesn't happen IRL with location bins but to me it's a bug, unless I didn't understood what was the purpose of the unreserve process in this scenario.

Reply:
Indeed the method should take in argument an option product. For the location content transfer, it is correct to unreserve all others moves as we empty the location. For this scenario where we move only a product, we should limit to the given product.

Reply:
Fixed here: https://github.com/OCA/wms/pull/482